src/extension-runners/chromium.js

```diff
@@ -127,8 +193,21 @@ export class ChromiumExtensionRunner {
       chromeFlags.push(...this.params.args);
     }
 
-    if (this.params.chromiumProfile) {
-      chromeFlags.push(`--user-data-dir=${this.params.chromiumProfile}`);
+    const profileParts =
+      await ChromiumExtensionRunner.getProfileParts(this.params);
+
+    if (profileParts.userDataDir
+      && profileParts.profileDirName
+      && !this.params.keepProfileChanges
+    ) {
+      // copy profile dir to this temp user data dir.
+      fs.copySync(this.params.chromiumProfile, path.join(
```

Comment:
Let's use `await fs.copy`, as there is no real need for this to be synchronous.

> The use of fs-extra is now reduced, and we could reevaluate if it is still justified.

There are multiple (dev) dependencies with overlapping functionality. For example, `mz` and `fs-extra` both offer promisified `fs` methods, and `fs-extra` and `copy-dir` both offer ways to copy. I think that `fs-extra` is the best replacement in the long term.